src/app/components/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,39 @@
+import { FC } from "react";
+import { TextInputProps } from "./TextInput.types";
+import styles from "./TextInput.module.css";
+
+export const TextInput: FC<TextInputProps> = ({ testId, type, required, label, value, error, helperText, placeholder, disabled, onChange, id, name}) => {
+
+  return (
+    <div
+      className={styles.textInput}
+      data-testid={testId}
+    >
+      <label
+        className={`${!!error ? styles.textInputLabelError : ""} ${styles.textInputLabel}`}
+        htmlFor={id}>
+        {label}
+        {required && <span className={`${!!error ? styles.requiredError : ""} ${styles.required}`}> *</span>}
+      </label>
+      <div className={styles.textInputContainer}>
+        <input
+          type={type}
+          name={name}
+          id={id}
+          aria-label={required ? `${label} *` : label}
+          aria-describedby={error ? `${testId}-error` : `${testId}-helper`}
```

Comment:
testId is used only for testing purposes and it should not be used in the prod code.